src/libraries/System.Net.Sockets/tests/FunctionalTests/ArgumentValidationTests.cs

```diff
@@ -505,7 +505,7 @@ public void Select_LargeList_Throws_ArgumentOutOfRange()
         [Fact]
         public void AcceptAsync_NullAsyncEventArgs_Throws_ArgumentNull()
         {
-            Assert.Throws<ArgumentNullException>(() => GetSocket().AcceptAsync(null));
+            Assert.Throws<ArgumentNullException>(() => GetSocket().AcceptAsync((SocketAsyncEventArgs)null));
```

Comment:
Presumably this cast is to resolve some ambiguity the compiler started complaining about?  What was it?

Reply:
Ah.  I'm guessing previously this would always bind to the SocketsAsyncEventArgs instance, because once the instance AcceptAsync was found, the compiler wouldn't consider the extensions.  But now there's also the AcceptAsync that takes a Socket argument, and it doesn't know which we mean.

Technically this is a source breaking change, but I think it's ok.  We've accepted such API additions in the past.

Reply:
The ambiguity is when you call AcceptAsync(null). Previously this always resolved to the SAEA overload. Now it's ambiguous between that and the newly-added instance method `AcceptAsync(Socket? acceptSocket)`.

Technically, this is a breaking change I suppose -- but previously, calling AcceptAsync(null) would always throw ArgumentNullException. Now it will give a compiler error about ambiguity. I think that's fine, but let me know if you disagree.

Reply:
Yup :)